cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/sinks/NGSIMySQLSink.java

```diff
@@ -406,25 +410,32 @@ public String getValuesForInsert() {
             
                 while (it.hasNext()) {
                     String entry = (String) it.next();
-                    ArrayList<String> values = (ArrayList<String>) aggregation.get(entry);
-                    String value = values.get(i);
+                    ArrayList<JsonElement> values = (ArrayList<JsonElement>) aggregation.get(entry);
+                    JsonElement value = values.get(i);
+                    String stringValue = null;
                     if (attrNativeTypes) {
                         LOGGER.debug("[" + getName() + "] aggregation entry = "  + entry );
-                        if (value == null || value.equals("")) {
-                            value = "NULL";
+                        if (value.isJsonNull()) {
+                            stringValue = "NULL";
+                        } else if (value.isJsonPrimitive()) {
+                            if (value.getAsJsonPrimitive().isBoolean()) {
+                                stringValue = value.getAsString();
```

Comment:
Needed to convet to uppercase?

Reply:
Fixed in [a79f96e](https://github.com/telefonicaid/fiware-cygnus/pull/1784/commits/a79f96e3c8c14b278edae5c051abc533cb1d263d)